lib/ansible/plugins/callback/foreman.py

```diff
@@ -80,154 +80,169 @@
 from ansible.plugins.callback import CallbackBase
 
 
+def build_log(data):
+    for source, msg in data:
+        if 'failed' in msg:
+            level = 'err'
+        elif 'changed' in msg and msg['changed']:
+            level = 'notice'
+        else:
+            level = 'info'
+
+        yield {
+            "log": {
+                'sources': {
+                    'source': source,
+                },
+                'messages': {
+                    'message': json.dumps(msg),
+                },
+                'level': level,
+            }
+        }
+
+
+def get_time():
+    """
+    Return the time for measuring duration. Prefers monotonic time but falls back to the regular
+    time on older Python versions.
+    """
+    try:
+        return time.monotonic()
+    except AttributeError:
+        return time.time()
+
+def get_now():
+    """
+    Return the current timestamp as a string to be sent over the network.
+    """
+    return datetime.utcnow().isoformat()
```

Comment:
Is it the same for Python 2 and Python 3?

Reply:
Yes they are the same, but not actually what I intended. Looks like `utcnow()` returns a naive datetime object so `isoformat()` doesn't actually output a timezone. Python 3 does have a `datetime.timezone.utc` but I don't see an easy equivalent in Python 2 so it's probably easier to use a manual format which includes `Z` to indicate it's UTC.

Reply:
smth like `datetime.utcnow().replace(tzinfo=tz).isoformat()` maybe

Reply:
I considered that, but I don't think pytz is a dependency you can count on and in Python 2 you don't have `datetime.timezone.utc`. The following could work, but is rather verbose:
```python
try:
  from datetime import timezone
except ImportError:  # Python 2
  return datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.%fZ')
else:
  return datetime.utcnow().replace(tzinfo=timezone.utc).isoformat()
```

Reply:
You may also be interested in my comment about UTC timezones here: https://github.com/ansible/ansible/pull/54792#discussion_r273107955